xls/tools/run_timing_characterization.py

```diff
@@ -0,0 +1,178 @@
+#
+# Copyright 2020 The XLS Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Runs the timing_characterization with OpenROAD flow-scripts tooling and libraries."""
+# Assume that https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts are build
+# ALSO assume that OpenROAD has pre-process libraries
+import subprocess
+import datetime
+import random
+import os
+
+from absl import app
+from absl import logging
+from absl import flags
+from time import sleep
+
+
+class WorkerConfig:
```

Comment:
This should probably be a dataclass. Example use:

https://github.com/google/xls/blob/18742d56a5089329a9eaacb15f186b385e1e167b/xls/fuzzer/run_fuzz_multiprocess.py#L38

You can add default values for some field (all of them) if you don't want to explicitly specify them on construction.